### ISSUE TITLE
AMD 68xx series GPU stops hashing but does not report lower hashrate

### ISSUE THREAD

**Issue by KillrBee** [sp-hash/TeamBlackMiner#271]
There's a problem where the GPU fails and keeps the hashrate and power draw values the same before the miner recognizes it to attempt and fail a restart. Once detected the miner will drop the reported hashrate to 0.00, but it will  keep the original power draw value, before finally failing and restarting the miner (often times 20+ minutes later).  

```
00:08:40 [2022-03-01 12:28:38.305] GPU0    gfx1030  OpenCL  0       190     35/0    80      1360    1069    124    
00:08:41 [2022-03-01 12:29:42.141] GPU0    gfx1030  OpenCL  0       190     35/0    80      1360    1069    124    
00:08:42 [2022-03-01 12:30:45.077] GPU0    gfx1030  OpenCL  0       190     35/0    80      1360    1069    124    
00:08:43 [2022-03-01 12:31:48.010] GPU0    gfx1030  OpenCL  0       190     35/0    80      1360    1069    124    
00:08:44 [2022-03-01 12:32:52.761] GPU0    gfx1030  OpenCL  0       190     34/0    80      1360    1069    124    
00:08:45 [2022-03-01 12:33:52.011] GPU0    gfx1030  OpenCL  0       190     24/0    80      1360    1069    124    
00:08:46 [2022-03-01 12:35:01.603] GPU0    gfx1030  OpenCL  0       190     21/0    80      1360    1069    124    
00:08:47 [2022-03-01 12:36:01.020] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    123    
00:08:48 [2022-03-01 12:37:01.040] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    123    
00:08:49 [2022-03-01 12:38:07.730] GPU0: has timed out, trying to restart..
00:08:49 [2022-03-01 12:38:07.730] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    123    
00:08:50 [2022-03-01 12:39:07.060] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    123    
00:08:51 [2022-03-01 12:40:07.033] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    122    
00:08:52 [2022-03-01 12:41:13.875] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    122    
00:08:53 [2022-03-01 12:42:13.044] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    122    
00:08:54 [2022-03-01 12:43:13.072] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    122    
00:08:55 [2022-03-01 12:44:13.088] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    122    
00:08:56 [2022-03-01 12:45:18.780] GPU0    gfx1030  OpenCL  0       190     19/0    80      1360    1069    121    
00:08:57 [2022-03-01 12:46:18.097] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    121    
00:08:58 [2022-03-01 12:47:25.391] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    121    
00:08:59 [2022-03-01 12:48:25.013] GPU0    gfx1030  OpenCL  0       190     20/0    80      1360    1069    121    
00:09:00 [2022-03-01 12:49:25.082] GPU0: GPU thread crashed exiting program.

```
If you look at the temperatures, you can see the miner stops mining at 8 hours and 44 minutes, but doesn't report it until 8h49m at which point the miner will report a 0.00 MH/s value but continue mining and keeping the reported power-level (while the temperature drops to ambient-idle).

Putting a watchdog on overall hashrate is difficult because there are various other factors that may induce a transient -60 MH/s rate, while power draw for a functioning rig will always be relatively consistent.  
I realize that amdgpu reporting might be giving the incorrect power draw, so it may not be simple to accurately report power, however, Ideally if the miner can't recover a GPU within a few minutes, it should attempt to restart.  In the above case it was 15 minutes (2 cards failed simultaneously), before the miner restarted which costs more lost $$ than a full miner restart after a few minutes of a GPU being down.

**Comment by KillrBee** [comment]
> If you push the card too hard, or under volt too much the cards will crash. We will try to improve the crash detection in the next release and exit quicker.

Yeah, but it mined for almost 9 hours before crashing -- temperatures on the cards are low and I'm not pushing the GPU or Memory to any extremes, and any undervaluing issues should show up earlier than 9 hours in...?

**Comment by sp-hash** [comment]
fixed in v1.58
